addons/crm/wizard/crm_lead_lost.py

```diff
@@ -10,20 +10,20 @@ class CrmLeadLost(models.TransientModel):
     _name = 'crm.lead.lost'
     _description = 'Get Lost Reason'
 
+    lead_ids = fields.Many2many('crm.lead')
     lost_reason_id = fields.Many2one('crm.lost.reason', 'Lost Reason')
     lost_feedback = fields.Html(
         'Closing Note', sanitize=True
     )
 
     def action_lost_reason_apply(self):
         self.ensure_one()
-        leads = self.env['crm.lead'].browse(self.env.context.get('active_ids'))
```

Comment:
For backward compatibility, we could have an override of default_get setting active_ids or [active_id] to default_lead_ids (if not already set), if active_model is crm.lead . Does not cost much and could ease some usage...

Reply:
Maybe yeah, added a short block for that

Reply:
Thinking of it, maybe we should drop that code ...  anyway the server action won't work anymore (python code disappeared) and the act_window is correctly updated. So I suppose we can remove the code: either we keep all for backward compatibility, either we clean everything (in favor of cleaning). What do you think ?

Reply:
Actually I guess it would work if we remove the action here without removing it in the upgrade script as it will keep its behaviour from 16.2, which is to call the act_window. And the act_window will get the active_ids from context.

I've added an upgrade PR, we should at least cancel the renaming since it's pointless now. We could just not delete the action as far as retro-compatibility goes, though it will surely break on its own at some point in the future if people use it as-is.

Reply:
But server action code calls 'action_set_lost_with_reason' which is removed ... so it won't work anymore post update, will it ?

Reply:
But server action code calls 'action_set_lost_with_reason' which is removed ... so it won't work anymore post update, will it ?

Reply:
It should because the change to calling `action_set_lost_with_reason` was only added in this minor version, meaning nobody is actually using that right now, nor will they ever. 

In 16.2 it calls either `lead.action_set_lost` or the act window directly from the server action. Since both still exist, and since the act window will retrieve the active_ids and feed it into default_lead_ids, it should work fine.

https://github.com/odoo/odoo/blob/saas-16.2/addons/crm/data/ir_action_data.xml#L14

Reply:
Yep, would have been confusing. I've added a foreword to say that commit is cancelled out by this one in the same minor version, and justified why we didn't remove the action in upgrade (assuming we're ok with not removing it in upgrade...)

Reply:
It is ok to keep it at upgrade, as it still works. It is just not needed anymore in standard codebase. I like that we keep default data as light as possible for new DBs ^^